src/queue.js

```diff
@@ -14,9 +14,13 @@ import {
   SQSClient,
   SendMessageCommand,
 } from '@aws-sdk/client-sqs';
+import { randomUUID } from 'crypto';
+// eslint-disable-next-line no-unused-vars
+import { BlobStorage } from './blob-storage.js';
 
 /**
  * @typedef QueueConfig
+ * @property {BlobStorage} [blobStorage]
  * @property {SQSClient} [client]
  * @property {Object} credentials
```

Comment:
For reused typedefs, it's possible to put them in a separate file. For example, creating a `typedefs.js` file:

```

/**
 * @typedef {AwsCredentials}
 * @property {string} accessKeyId
 * @property {string} accessKeySecret
 */

export {}; // this is important to ensure intellisense works
```

Then it can be imported:

```
/**
 * @typedef QueueConfig
 * @property {import('./typedefs').AwsCredentials} credentials
 */
```

Up to you, but this is something I've done in the past to avoid needing to define the same typedefs over and over. Intellisense works in VSCode, at least.